src/wallet/rpc/backup.cpp

```diff
@@ -916,6 +916,7 @@ static std::string RecurseImportData(const CScript& script, ImportData& import_d
         return "unrecognized script";
     } // no default case, so the compiler can warn about missing cases
     CHECK_NONFATAL(false);
+    return ""; // not reached
```

Comment:
Hmm, I am wondering if it makes sense to introduce a macro that throws `NonFatalCheckError` unconditionally?

`CHECK_NONFATAL(false)` is inspired by `assert(false)` to mark unreachable code. Though, hiding the implementation behind a function call makes it harder for the compiler to realize the function never returns when called with `false`. (The same happens when using `Assert(false)`).

Reply:
How about a macro `UNREACHABLE_NONFATAL()` and `UNREACHABLE()`?

Reply:
>Hmm, I am wondering if it makes sense to introduce a macro that throws NonFatalCheckError unconditionally?

CHECK_NONFATAL(false) is inspired by assert(false) to mark unreachable code. Though, hiding the implementation behind a function call makes it harder for the compiler to realize the function never returns when called with false. (The same happens when using Assert(false)).

Concept ACK, this will make the code much cleaner